flow/project.py

```diff
@@ -1048,12 +1049,8 @@ def _generate_id(self, jobs, operation_name=None, index=0):
             concat_jobs_str = str(jobs[0])[0:8]
 
         separator = getattr(project._environment, 'JOB_ID_SEPARATOR', '/')
-        readable_name = '{project}{sep}{jobs}{sep}{op_string}{sep}{index:04d}{sep}'.format(
-                    sep=separator,
-                    project=str(project)[:12],
-                    jobs=concat_jobs_str,
-                    op_string=op_string[:12],
-                    index=index)[:max_len]
+        readable_name = f'{str(project)[:12]}{separator}{concat_jobs_str}{separator}' \
```

Comment:
We print the length first in `_JobOperation.__str__`. I think that order makes sense here too.